openslides/core/static/js/core/site.js

```diff
@@ -525,33 +525,87 @@ angular.module('OpenSlidesApp.core.site', [
 .controller('SearchBarCtrl', [
     '$scope',
     '$state',
-    function ($scope, $state) {
-        $scope.search = function(query) {
-            $scope.query = '';
+    '$sanitize',
+    function ($scope, $state, $sanitize) {
+        $scope.search = function() {
+            var query = $sanitize($scope.query);
+            $scope.query = "";
             $state.go('search', {q: query});
         };
     }
 ])
+
 // Search Controller
 .controller('SearchCtrl', [
     '$scope',
     '$http',
     '$stateParams',
+    '$location',
+    '$sanitize',
     'DS',
-    function ($scope, $http, $stateParams, DS) {
+    function ($scope, $http, $stateParams, $location, $sanitize, DS) {
+        $scope.fullword = false;
+        $scope.filterAgenda = true;
+        $scope.filterMotion = true;
+        $scope.filterAssignment = true;
+        $scope.filterUser = true;
+        $scope.filterMedia = true;
+
         // search function
-        $scope.search = function(query) {
-            $http.get('/core/search_api/?q=' + query).then(function(success) {
-                var elements = success.data.elements;
-                $scope.results = [];
-                angular.forEach(elements, function(element) {
-                    DS.find(element.collection, element.id).then(function(data) {
-                        data.urlState = element.collection.replace('/','.')+'.detail';
-                        data.urlParam = {id: element.id};
-                        $scope.results.push(data);
+        $scope.search = function(q) {
```

Comment:
I do not understand because I think in this case there should be something like `search(<something here>)` somewhere in a template. I can not find such a string in one of our templates. If there is some hidden magic for this function call then we can merge this. No problem.